File: python/NeetCode150/13_max_area.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights) - 1
        max_area = 0
        while(l < r):
            mx = max(heights[l], heights[r])
            mn = min(heights[l], heights[r])
            area = mn * (r - l)
            if area > max_area:
                max_area = area
            if heights[r] > heights[l]:
                l += 1
            else:
                r -= 1
        return max_area
        

File: java/NeetCode150/13_max_area.java
class Solution 
{
    public int maxArea(int[] heights) 
    {
        int l = 0;
        int r = heights.length - 1;
        int max_area = 0;
        while(l < r)
        {
            int area = Math.min(heights[l], heights[r]) * (r - l);
            max_area = Math.max(max_area, area);
            if(heights[l] < heights[r])
            {
                l += 1;
            }
            else
            {
                r -= 1;
            }
        }
        return max_area;
    }
}
